common/src/main/java/dev/latvian/mods/kubejs/item/custom/MultitoolItemJS.java

```diff
@@ -0,0 +1,179 @@
+import dev.latvian.mods.kubejs.registry.KubeJSRegistries;
+import dev.latvian.mods.kubejs.typings.Info;
+import dev.latvian.mods.kubejs.typings.Param;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+import net.minecraft.world.item.DiggerItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.math.RoundingMode.CEILING;
+
+public class MultitoolItemJS extends DiggerItem {
+    public static class Builder extends HandheldItemBuilder {
+        protected boolean isAxe = false, isHoe = false, isPickaxe = false, isShovel = false;
+        protected final List<Float> speedValues = new ArrayList<>(5);
+        protected final List<Float> attackValues = new ArrayList<>(5);
+        protected final Set<TagKey<Block>> mineableTags = new HashSet<>();
+        public Builder(ResourceLocation i) {
+            super(i, 0, 0);
+        }
+        @Info(value = """
+            Adds a tool to the multi-tool.
+            
+            Valid tool types include: 'axe', 'hoe', 'pickaxe', and 'shovel'.
+            """, params =
+        @Param(name = "tool", value = "The name of the tool to add to the multi-tool. Will error if it is an unknown tool type."))
+        public Builder tool(String tool) {
+            // maybe swords and shears (after #673 is merged)?
+            return switch (tool) {
+                default -> throw new IllegalArgumentException("Unknown tool type '" + tool +
+                        "', valid tool types are: 'axe', 'hoe', 'pickaxe', and 'shovel'!");
+                case "axe" -> {
+                    isAxe = true;
+                    yield addValues(6, -3.1f, BlockTags.MINEABLE_WITH_AXE);
+                }
+                case "hoe" -> {
+                    isHoe = true;
+                    yield addValues(-2, -1, BlockTags.MINEABLE_WITH_HOE);
+                }
+                case "pickaxe" -> {
+                    isPickaxe = true;
+                    yield addValues(1, -2.8f, BlockTags.MINEABLE_WITH_PICKAXE);
+                }
+                case "shovel" -> {
+                    isShovel = true;
+                    yield addValues(1.5f, -3, BlockTags.MINEABLE_WITH_SHOVEL);
+                }
+            };
+        }
```

Comment:
Should this be an Enum?

Reply:
if an enum was the arg, would rhino convert a string into the enum value?
bcuz if not, i dont really understand why this would need to be an enum or how it would work as an enum

Reply:
yes, rhino autowraps strings to enums. You could also then use an EnumSet instead of seperate booleans for each tool type, and have the addValues function be function on the enum, which I think would completely remove the need for a switch statement.